### ISSUE TITLE
issues about preparing the data and training

### ISSUE THREAD

**Issue by Moreland-cas** [BestJuly/VCP#1]
Good repo! 
When I was training the network, I found that in the data preparation process, the RandomCrop used for several frames in a video segment was not consistent, which caused the background and characters to shake. At the same time, I also suffered from the problem of model not fitting. Is there any way to solve it?

**Comment by BestJuly** [comment]
Hi, @2017K8009929048 

For the RandomCrop part, I do not think the background and characters should shake. Because though RandomCrop will randomly crop frames, it is controlled by random seed.
```
# in ucf101.py
for frame in clip:
    random.seed(seed)
    frame = self.toPIL(frame) # PIL image, this is activated when using skvideo.io
    frame = self.transforms_(frame) # tensor [C x H x W]
```
For each video clip, the random seed is controlled by the code using a fixed seed.

By the way, I noticed that the [newest pytorch version](https://pytorch.org/docs/stable/torchvision/transforms.html) has replaced `random.seed(seed)` with `torch.manual_seed(seed)`. Therefore, if you find shaking, you can use `torch.manual_seed(seed)` instead.

For the problem that the model is not fitting, could you show more details? Such as in which training period you met the problem, or using which command. And also please check whether the only change is the dataset path.

**Comment by Moreland-cas** [comment]
> Hi, @2017K8009929048
> 
> For the RandomCrop part, I do not think the background and characters should shake. Because though RandomCrop will randomly crop frames, it is controlled by random seed.
> 
> ```
> # in ucf101.py
> for frame in clip:
>     random.seed(seed)
>     frame = self.toPIL(frame) # PIL image, this is activated when using skvideo.io
>     frame = self.transforms_(frame) # tensor [C x H x W]
> ```
> 
> For each video clip, the random seed is controlled by the code using a fixed seed.
> 
> By the way, I noticed that the [newest pytorch version](https://pytorch.org/docs/stable/torchvision/transforms.html) has replaced `random.seed(seed)` with `torch.manual_seed(seed)`. Therefore, if you find shaking, you can use `torch.manual_seed(seed)` instead.
> 
> For the problem that the model is not fitting, could you show more details? Such as in which training period you met the problem, or using which command. And also please check whether the only change is the dataset path.

I used my own method to solve the problem of randomcrop, but it did not solve the problem of model not fitting.
![train](https://user-images.githubusercontent.com/48342903/109447333-64c48200-7a7e-11eb-8b9b-e99f4e56cc6b.jpg)
This is the log of my training train_vcop.py. Loss dropped to around 1.79 and no longer dropped. Is the number of epochs I trained too few? Can you provide your training log or tensorboard visualization?

**Comment by BestJuly** [comment]
Your log style is different from this repo, and you metioned this is the log when you used `train_vcop.py`. The method VCOP is a different method and I am not sure whether your problem lies to the VCOP because in my repo, I just try to reproduce the VCP method and the training (e.g. data preparation) is different from VCOP.

I do not have `train_vcop.py` in this repo but `train_vcp.py`. Usually, it should start to converge very fast (less than 30 epochs). Currently, your reported logs seem to be random guess.

For the training logs, because this repo is just a re-implementation and it was done one year ago. I am afraid that I could not find previous logs. If you do not change anything except for the data path and use the same pytorch version as I did, you should achieve comparable results as I reported in the Readme.md.
